### PR TITLE
add browser platform

### PR DESCRIPTION
**Pull request by habibillah**
Add browser platform

### CODE REVIEW

**Comment by yxliang01** [comment]
@EddyVerbruggen This pr is fantastic if it works! Could you merge it or leave a comment here if anything u don't like?

**Comment by EddyVerbruggen** [comment]
@yxliang01 If you can confirm it works as expected then I will merge it. You can add https://github.com/habibillah/cordova-plugin-googleplus/ to your project to test it.

Thanks guys!

**Comment by habibillah** [comment]
@yxliang01 See this doc references
- https://github.com/EddyVerbruggen/cordova-plugin-googleplus/pull/414/files#diff-04c6e90faac2675aa89e2176d2eec7d8R57
- https://github.com/EddyVerbruggen/cordova-plugin-googleplus/pull/414/files#diff-04c6e90faac2675aa89e2176d2eec7d8R123

**Comment by paholicsgabor** [comment]
Hi there,
I am developing a web/android/ios app with ionic framework. I try to keep the codebase same for all platforms. Now I am about to solve google sign in on web platform and I have 2 options:
- split the codebase 
- help to test this pr.

I prefer the second solution, but I have some problems. I am not sure how to add this branch to my ionic project. Ionic is using
"@ionic-native/google-plus": "4.4.0"
which is using
"cordova-plugin-googleplus": "5.1.1",
(copied from package.json)

Can anybody help me how to test this branch in ionic framework? 

Thanks for your help.

**Comment by sagivf** [comment]
Checked this on the browser (not yet on ios and android) and works great.
Is there any way to avoid the popup window? Perhaps using redirects instead?

**Comment by lucaszamboulis** [comment]
@EddyVerbruggen @habibillah Just tried this, works as expected.

File: src/browser/GooglePlusProxy.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+                for (var i = 0; i < __googleCallbacks.length; i++) {
+                    __googleCallbacks[i].call(GooglePlusProxy);
+                }
+
+                // Listen for sign-in state changes.
+                gapi.auth2.getAuthInstance().isSignedIn.listen(GooglePlusProxy.updateSigninStatus);
+            }, function(error) {
+                if (error.details) {
+                    console.error(error.details);
```
Comment (sprt) [inline src/browser/GooglePlusProxy.js]: This always trigger this error for me, even though `WEB_APPLICATION_CLIENT_ID` is set in `config.xml`.

> Not a valid origin for the client: http://localhost:8000 has not been whitelisted for client ID null. Please go to https://console.developers.google.com/ and whitelist this origin for your project's client ID.

**Comment by habibillah** [comment]
as mentioned in error message, you should white list http://localhost:8000/
in google developer console.

**Comment by sprt** [comment]
I've done that, the problem is the WEB_APPLICATION_CLIENT_ID isn't parsed correctly for some reason, it's set to `null`.

> [...] not been whitelisted for client ID **null** [...]

**Comment by habibillah** [comment]
Have you tried Cordova prepare command?

**Comment by sprt** [comment]
Yup, that then `build` and `run android`.

**Comment by sprt** [comment]
Nevermind, I redid the whole thing and it's working now. :+1:

File: plugin.xml [hunk 2]
```diff
@@ -63,6 +63,7 @@
     <source-file src="src/ios/GooglePlus.m"/>
 
     <preference name="REVERSED_CLIENT_ID" />
+    <preference name="WEB_APPLICATION_CLIENT_ID" />
```
Comment (EddyVerbruggen) [inline plugin.xml]: Shouldn't this be in the `<browser>..</browser>` section of plugin.xml? At the moment it's in the `<ios>..</ios>` section.

Comment (habibillah) [reply plugin.xml]: you are correct. that should inside `browser` for more readability. thanks. I updated it

**Comment by Daskus1** [comment]
Great work! Is this going to be merged anytime soon?

**Comment by EddyVerbruggen** [comment]
@Daskus1 soon enough? 😎

Thanks a lot, @habibillah!

**Pull request merged**